src/IrcClient/IrcClient.cs

```diff
@@ -1258,6 +1280,48 @@ private string _NextNickname()
             }
             return NicknameList[_CurrentNickname];
         }
+
+        private string _UnescapeTagValue(string tagValue)
+        {
+            int lastPos = 0;
+            int pos = 0;
+            string sequence;
+            string unescaped = "";
```

Comment:
This should probably be a StringBuilder to reduce GC pressure (strings are immutable and thus creates a new object for every concat)